preflight-checks/nccl-allreduce/nccl_allreduce/tests/test_gang.py

```
@@ -0,0 +1,172 @@
+from nccl_allreduce.gang import GangConfig, GangConfigReader, GangWaiter, PeerInfo
+
+
+def write_configmap(tmpdir: str, data: dict[str, str]) -> None:
+    """Write ConfigMap-style files into a directory."""
+    for key, value in data.items():
+        with open(os.path.join(tmpdir, key), "w", encoding="utf-8") as f:
+            f.write(value)
+
+
+class TestGangConfigReader:
+    def test_read_parses_configmap(self, tmp_path: str) -> None:
```

Ruff (0.15.2)

[warning] Missing docstring in public method

(D102)